lib/rubocop/cop/i18n/rails_i18n/decorate_string.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module I18n
+      module RailsI18n
+        # This cop is looks for strings that appear to be sentences but are not decorated.
+        # Sentences are determined by the STRING_REGEXP. (Upper case character, at least one space,
+        # and sentence punctuation at the end)
+        #
+        # @example
+        #
+        #   # bad
+        #
+        #   "Result is bad."
+        #
+        # @example
+        #
+        #   # good
+        #
+        #   t("result_is_good")
+        #   I18n.t("result_is_good")
+        #
+        class DecorateString < Cop
+          STRING_REGEXP = /^\s*[[:upper:]][[:alpha:]]*[[:blank:]]+.*[.!?]$/.freeze
+          SUPPORTED_DECORATORS = %w[
+            t
+            t!
+            translate
+            translate!
+          ].freeze
+
+          def on_dstr(node)
+            check_for_parent_decorator(node) if dstr_contains_sentence?(node)
+          end
+
+          def on_str(node)
+            return unless sentence?(node)
+
+            parent = node.parent
+            if parent.respond_to?(:type)
+              return if parent.regexp_type? || parent.dstr_type?
+            end
+
+            check_for_parent_decorator(node)
+          end
+
+          private
+
+          def sentence?(node)
+            child = node.children[0]
+            if child.is_a?(String)
+              if child.valid_encoding?
+                child.encode(Encoding::UTF_8).chomp =~ STRING_REGEXP
```

Comment:
Due to my lack of knowledge, I am trying some out in IRB but I shouldn't be blocking you from submitting (Joel already approved it so no block here).  Does this cover string interpolation case or that is outside of scope? I am just asking.

Reply:
`dstr` (from what Joe and I read) are interpolated strings, so it should work for them without issue. I can add a test case for it though. However, this code was more or less just copied from the original GetText implementation.